tests/deployments/helm/helm_test.py

```diff
@@ -192,40 +209,60 @@ def install_helm_chart(k8s_cluster, values_path):
         print(k8s_cluster.exec_kubectl(f"describe daemonset {daemonset_name}", namespace=k8s_cluster.test_namespace))
 
 
-def test_helm(k8s_cluster):
-    with k8s_cluster.create_resources([APP_YAML_PATH]), tiller_rbac_resources(
-        k8s_cluster
-    ), fake_backend.start() as backend:
-        init_helm(k8s_cluster)
-
-        with k8s_cluster.run_tunnels(backend) as proxy_pod_ip:
-            with release_values_yaml(k8s_cluster, proxy_pod_ip, backend) as values_path:
-                install_helm_chart(k8s_cluster, values_path)
-                try:
-                    assert wait_for(
-                        p(
-                            has_datapoint,
-                            backend,
-                            dimensions={"container_name": "prometheus", "application": "helm-test"},
-                        ),
-                        timeout_seconds=60,
-                    )
-                    assert wait_for(
-                        p(has_datapoint, backend, dimensions={"plugin": "signalfx-metadata"}), timeout_seconds=60
-                    )
-                finally:
-                    for pod in get_pods_by_labels("app=signalfx-agent", namespace=k8s_cluster.test_namespace):
-                        print("pod/%s:" % pod.metadata.name)
-                        status = exec_pod_command(
-                            pod.metadata.name, AGENT_STATUS_COMMAND, namespace=k8s_cluster.test_namespace
+def get_kubeconfig(k8s_cluster, cont):
+    args = f"kubectl --kubeconfig={k8s_cluster.kube_config_path} config view --raw --flatten"
+    proc = subprocess.run(args, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, encoding="utf-8")
+    assert proc.returncode == 0, f"{args}:\n{proc.stdout}"
+    copy_file_content_into_container(proc.stdout, cont, k8s_cluster.kube_config_path)
+
+
+@contextmanager
+def run_helm_image(k8s_cluster, helm_version):
+    opts = dict(path=REPO_ROOT_DIR, dockerfile=SCRIPT_DIR / "Dockerfile", buildargs={"VERSION": helm_version})
+    with run_service("helm", **opts) as cont:
+        get_kubeconfig(k8s_cluster, cont)
+        yield cont
+
+
+@pytest.mark.parametrize("helm_version", ["2.15.0", "3.0.0"])
+def test_helm(k8s_cluster, helm_version):
+    helm_major_version = int(helm_version.split(".")[0])
+    with run_helm_image(k8s_cluster, helm_version) as cont:
+        with k8s_cluster.create_resources([APP_YAML_PATH]), tiller_rbac_resources(
```

Comment:
If it is 3.0, we really shouldn't be installing anything related to Tiller if it was removed.

Reply:
> If it is 3.0, we really shouldn't be installing anything related to Tiller if it was removed.

I updated the tests for both helm v2 and v3.